auth/credentials/idtoken/idtoken.go

```diff
@@ -113,13 +113,20 @@ func NewCredentials(opts *Options) (*auth.Credentials, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
-	if b := opts.jsonBytes(); b != nil {
-		return credsFromBytes(b, opts)
-	}
-	if metadata.OnGCE() {
+	b := opts.jsonBytes()
+	if b == nil && metadata.OnGCE() {
 		return computeCredentials(opts)
+	} else if b == nil {
+		creds, err := credentials.DetectDefault(&credentials.DetectOptions{
+			Audience: opts.Audience,
```

Comment:
This aud is for the idtoken, not the creds. This should be some default scopes

Reply:
Fixed, thanks!